server/src/main/java/org/elasticsearch/plugins/ExtensiblePlugin.java

```diff
@@ -27,8 +29,22 @@
  */
 public interface ExtensiblePlugin {
 
+    interface ExtensionLoader {
+        /**
+         * Load extensions of the type from all extending plugins. The concrete extensions must have either a no-arg constructor
+         * or a single-arg constructor accepting the specific plugin class.
+         * @param extensionPointType the extension point type
+         * @param <T> extension point type
+         * @return stream of all implementing extensions.
+         */
+        <T> Stream<T> loadExtensions(Class<T> extensionPointType);
```

Comment:
Why a stream? It seems to lend itself to longer lines, but not easier to read code in this case, especially in the implementation below

Reply:
I changed it to a List. Stream is nice though, since its contract is read-only and read-once - and still easy to use (unlike Iterable and Iterator).